docking.js

```diff
@@ -544,7 +545,10 @@ var DockedDash = GObject.registerClass({
         ], [
             settings,
             'changed::intellihide',
-            this._updateVisibilityMode.bind(this)
+            () => {
+                this._updateVisibilityMode();
+                this._updateVisibleDesktop();
```

Comment:
In this case, if intellihide is disabled, shouldn't we reset the usable area?

It's not probably something that I'd like to do here though, but rather in the DockManager

Reply:
That's a good point. I tried to do that, and the result was bad, because there seems to be a delay between the change notification in DockManager object and the notification in each dock (that, or the callback is being called before in the DockManager object than in the docks themselves), so it fails.

Anyway, since the margin values are the same in both cases, it is not needed, because when intellihide is disabled, the margin value will be the same than in the WorkingArea data, and the margin values are presumed to be from the screen border, not from the WorkingArea border.

Reply:
Oh, ok... So you use the monitor size... Mh, this could be ok, even though I assume there could be problem in case the dock si then resized.

I'm wondering if we can just trigger a reset in a clean way in this case... Maybe just connect on the `changed::intellihide` in dockmanager and when it's false, just sending a reset?